### PR TITLE
Style/theme/brand config with example

### PR DESCRIPTION
**Pull request by namick**
## Changes

  - Add a `useBrandConfig` hook to allow the customization of components in this library.  We will have a basic style and theme, but both can be customized by passing config into the provider:
  ```<BrandConfigProvider style={customStyle} theme={customTheme} />```
  - Add a `Text` component that shows this functionality

TODO:
  - Rename and finish the theme customization
  - Add a generic type for customizing the brand config
  - Add a storybook story
  - Documentation

### CODE REVIEW

File: src/hooks/useLifeOmicConfig.tsx
Comment (markdlv) [inline src/hooks/useLifeOmicConfig.tsx:5]: Can this be `BrandConfig` or something like that?

File: src/hooks/useLifeOmicConfig.tsx
Comment (markdlv) [inline src/hooks/useLifeOmicConfig.tsx:41]: Let's stick with `useAppConfig` when we get to that hook - but this seems different.  Can this be `useBranding`?

File: src/theme/colors.ts
Comment (markdlv) [inline src/theme/colors.ts:1]: This looks great!  Great thoughts offline here.  It's important (to me) that this lib represents the "LifeOmic mobile app shell."  It should _not_ end up being the new Outcomes repo.  So ideally, these colors/spacing/etc. represent sensible defaults for us to develop components against.  And then we're overriding those in our Outcomes repo.

File: src/components/Text.test.tsx
Comment (sternetj) [inline src/components/Text.test.tsx:10]: Shouldn't need to add the provider here. It might be a good idea to remove it so we can ensure that themed components can be used with the default context value.

**Review by sternetj — APPROVED** [review]

File: src/components/Text.test.tsx
Comment (namick) [reply src/components/Text.test.tsx]: Oh, good call!

**Comment by loscm** [comment]
Requested review from a random `@lifeomic/security` team member, @indigocarmen, due to term: **shell**.

This review is not blocking and is for broader awareness.  Consider if this change requires deeper security review and ask when it is necessary.

- [Automation](https://github.com/lifeomic/pr-security-flag-scanner/blob/master/src/lambdas/pr-security-flag-scanner/index.ts) | [#pr-flagged-sec-review](https://lifeomic.slack.com/messages/pr-flagged-sec-review)

File: src/components/Text.tsx
Comment (markdlv) [inline src/components/Text.tsx:38]: Something that might be worth prototyping, as an additive alternative to components using `useBrandConfig` like this, is an HOC that captures:
* useBrandConfig usage, but
* is injected with a mapping between expected brandConfig props & the component's props

In my mind, that'd allow some flexibility like:
- While some components might be coupled to `useBrandConfig`, all don't have to be.  i.e. There's a "component purity" factor here in my mind
- Using regular RN components (or some other component lib) instead of ours.  This particularly comes to mind for components that aren't unique to LifeOmic, like this `Text` example

File: src/index.ts
Comment (markdlv) [inline src/index.ts:14]: I think you can back this change out.  `SecureStore` is already exported from `./common`

File: src/theme/colors.ts
Comment (markdlv) [reply src/theme/colors.ts:1]: I'm curious whether this folder should `defaultTheme`, `baseTheme` - something to that effect?  Esp. if it's wireframe-y, I'd think we always want/expect something layered on top of the default values provided in these files.

File: src/hooks/useBrandConfig.tsx
Comment (markdlv) [inline src/hooks/useBrandConfig.tsx:25]: So one can say `<BrandingConfigProvider styles={myStyleOverrides}>...` - very cool.  And eventually `theme` will also be injectable.  And those `styles` and `theme` props can be extensions of the types provided in this lib, with additional props etc.  Then if `useBrandConfig` had a generic, one could get back their own type.  Excited to see all this come together in storybook story/demo in some future PR

Comment (namick) [reply src/hooks/useBrandConfig.tsx:25]: Correct!  See `TODO:` list at top.  😄

File: src/hooks/useBrandConfig.tsx
Comment (markdlv) [inline src/hooks/useBrandConfig.tsx:11]: I'm eager to find/name strategies for ensuring this type doesn't balloon with props that we later want to refactor/remove as a breaking change.  The only I can think of are:
- Generally keep this type super simple and be cautious about adding to it (maybe need a comment)
- I wonder if we could separate out different groups into different types that could be (optionally) decorated/merged into a generic type used by this provider/context/hook?  Can explain further offline

**Review by markdlv — APPROVED** [review]
Lot of thoughts/ideas I commented in here, but removing `export * from './common/SecureStore';` is the only important one for this PR before merging.

File: src/hooks/useBrandConfig.tsx
Comment (namick) [reply src/hooks/useBrandConfig.tsx:11]: Agree 💯 . The idea here (in contrast to some ways we have done this in the past), would be that `styles:` property is composed from collections of styles imported from each component.  

All the style typing information should live with each component and be collected together here.  Might even be good to create a helper to build this up in the future if it gets to be too big here.

**Review by markdlv — APPROVED** [review]

**Pull request merged**
